src/anyio/_backends/_asyncio.py

```diff
@@ -1333,10 +1333,15 @@ async def wait_all_tasks_blocked() -> None:
 class TestRunner(abc.TestRunner):
     def __init__(self, debug: bool = False, use_uvloop: bool = True,
                  policy: Optional[asyncio.AbstractEventLoopPolicy] = None):
+        self._exceptions: List[Exception] = []
```

Comment:
```suggestion
        self._exceptions: List[BaseException] = []
```

?

Reply:
BaseExceptions crash the event loop; the handler is never called with such an exception.

Reply:
Cool